### ISSUE TITLE
Tile 22312 marked as unsure: B/R camera jump in sp6; bad trace shift fit

### ISSUE THREAD

**Issue by geordie666** [desihub/desisurveyops#58]
Tile 22313 from night 20220605 was marked as unsure:

https://data.desi.lbl.gov/desi/spectro/redux/daily/tiles/cumulative/22312/20220604/tile-qa-22312-thru20220604.png

Petal 6, in particular, has an excess of valid redshifts just above z~0.5 that seems suspicious.

D.S. reports that a B/R camera jump in sp7 makes the spectra look like this (hence the excess, as the 4000A break is likely confused with the jump):

<img width="579" alt="Screen Shot 2022-06-05 at 1 42 11 PM" src="https://user-images.githubusercontent.com/13952581/172070808-98cd4d54-d8d3-497d-9bc4-54203a9fc49e.png">
:

**Comment by julienguy** [comment]
My understanding is that the issue is in SP6. Sky fiber 3328 is bright in b6,  tile 22312 expid 00138142.
ra=245.6779 dec=32.3211 It's close to a bright star. It's not clear yet if this is the cause of an over subtraction of the sky.

**Comment by schlafly** [comment]
It's not clear to me how a bright fiber can lead to this ~symmetrical pattern across the b6 CCD.
![image](https://user-images.githubusercontent.com/1417841/172265090-02c11db3-dbba-4aff-b99e-50f539363762.png)
I tried to look at glow from the ion pumping but didn't see a corresponding peak in the pressure at the time this exposure was being observed.  And here we go from large effect to no effect in the time of one bright exposure, so I don't think this is the right explanation.

Because things look ~quadratic in fiber number, I'm now leaning toward weird PCA corrections & out of control TPCORR?  It's weird that the bright 5577A sky line looks uniformly ~undersubtracted in the sframe file, when I would have said that it ~has to be zeroed out by construction by TPCORR.  Wavelength solution could be really weird?

**Comment by julienguy** [comment]
- The error is in the throughput correction term that is entirely based on the relative intensity of one sky line for the blue CCDs.  
 - Disabling this correction solves the problem.
 - This incorrect variation of the sky line intensity is possibly due to an error in the wavelength calibration correction which also relies on the sky line, and is a quadratic function of the X coordinate in the CCD.
 - We could try re-extracting this frame with a change to the degree of the spectral trace shifts (with `desi_compute_trace_shifts --degyx 0` instead of `--degyx 2`).

**Comment by julienguy** [comment]
```
plot_fiber_traces -i /global/cfs/cdirs/desi/spectro/redux/daily/exposures/20220604/00138142/psf-b6-00138142.fits --other-psf /global/cfs/cdirs/desi/spectro/redux/daily/exposures/20220604/00138143/psf-b6-00138143.fits
```
![Screenshot from 2022-06-06 16-34-04](https://user-images.githubusercontent.com/5192160/172266102-e464d020-5269-4ae0-ac4d-e6f3f780eb1f.png)

**Comment by julienguy** [comment]
I confirm this is due to the trace shifts. This is a rare failure mode. We could think about changing the trace shift code. This failure will also be mitigated once we have in place a model for the throughput variations.

How to check this :
```

desi_compute_trace_shifts -i /global/cfs/cdirs/desi/spectro/redux/daily/preproc/20220604/00138142/preproc-b6-00138142.fits.gz --psf /global/cfs/cdirs/desi/spectro/redux/daily/calibnight/20220604/psfnight-b6-20220604.fits --degxx 2 --degxy 0 --degyx 0 --degyy 0 --sky --outpsf psf-b6-00138142-bis.fits

export OMP_NUM_THREADS=1
srun -N 1 -n 20 desi_extract_spectra -w 3600.0,5800.0,0.8 -i /global/cfs/cdirs/desi/spectro/redux/daily/preproc/20220604/00138142/preproc-b6-00138142.fits.gz -p psf-b6-00138142-bis.fits -o frame-b6-00138142-bis.fits --psferr 0.1 --barycentric-correction --mpi

desi_compute_sky -i frame-b6-00138142-bis.fits --fiberflat /global/cfs/cdirs/desi/spectro/redux/daily/exposures/20220604/00138142/fiberflatexp-b6-00138142.fits.gz -o sky-b6-00138142-bis.fits --adjust-wavelength --adjust-lsf --pca-corr /global/cfs/cdirs/desi/spectro/desi_spectro_calib/trunk/spec/sm7/skycorr-pca-sm7-b6.fits --fit-offsets

desi_process_exposure -i frame-b6-00138142-bis.fits --fiberflat /global/cfs/cdirs/desi/spectro/redux/daily/exposures/20220604/00138142/fiberflatexp-b6-00138142.fits.gz --sky ./sky-b6-00138142-bis.fits --no-tsnr --no-xtalk -o sframe-b6-00138142-bis.fits

plot_frame -i sframe-b6-00138142-bis.fits --sky-fibers --rebin 12
```
![Screenshot from 2022-06-06 16-53-25](https://user-images.githubusercontent.com/5192160/172267769-e3d01b41-e102-4f6b-ae1f-e57db11f935d.png)

**Comment by schlafly** [comment]
We think we now understand this.  @julienguy , what do we want to do about it?  Fix something, add it to the hypothetical list of tiles that get a special argument in their reductions, or mark it as bad?

**Comment by schlafly** [comment]
@julienguy, trying to clean up desisurveyops tickets during the shutdown---we planned to mark SP6 as bad for this tile and reprocess.  This is for a rare case where the raw data is probably fine but something tickles something in the trace shift calculation leading to a spurious B band trace shift fit on SP6.

Can you say that you agree with that approach?

**Comment by julienguy** [comment]
I don't think we want to change the code. A 2nd degree correction in the cross-dispersion direction works for all but one exposure. Let's mark just this exposure in b6 as bad.

**Comment by schlafly** [comment]
@akremin , can we mark b6 on 20220604/138142 tileid 22312 as bad and reprocess?

**Comment by schlafly** [comment]
@akremin, sorry for the whiplash, Julien has tabled work improving the trace shift fitting and so we're going to mark this as bad.  Can you please mark b6 on 20220604/138142 tileid 22312 as bad and reprocess?

**Comment by akremin** [comment]
Since this was a single exposure the standard star fitting fails without b6, so I've flagged the entire petal as bad for this exposure.

Specifically, I've set `BADCAMWORD` to a6 for 20220604/138142, tileid 22312 for both the svn `exposure_tables` and the daily versions (which are a snapshot of the daily processed data). I've removed the old reductions and rerun. I have also updated the tsnr table (`tiles-daily.csv`) for that night, as well as running `desi_night_qa` with `--recompute`.

The new data are now ready for QA.
